libs/community/langchain_community/document_compressors/flashrank_rerank.py

```diff
@@ -69,11 +71,10 @@ def compress_documents(
         final_results = []
 
         for r in rerank_response:
-            metadata = r["meta"]
-            metadata["relevance_score"] = r["score"]
-            doc = Document(
-                page_content=r["text"],
-                metadata=metadata,
-            )
-            final_results.append(doc)
+            if r["score"] >= self.score_threshold:
+                doc = Document(
+                    page_content=r["text"],
+                    metadata={"flashrank_rerank": {"id": r["id"], "relevance_score": r["score"]}, **r["metadata"]},
```

Comment:
does moving putting `relevance_score` inside `flash_rerank` represent a breaking change for users?

Reply:
To prevent overloading the already defined IDs in the metadata and to identify that this information is related to flashrank_rerank, we can avoid modifications to the metadata logic by adding a parameter prefix_metadata, if defined, gets added as a prefix to the ID and relevance_score. This way, you can ensure that the data is properly associated with flashrank_rerank without altering the existing metadata structure